rotkehlchen/db/upgrades/v36_v37.py

```diff
@@ -185,33 +187,131 @@ def _update_ens_mappings_schema(write_cursor: 'DBCursor') -> None:
     log.debug('Exit _update_ens_mappings_schema')
 
 
-def _fix_kraken_eth2_events(write_cursor: 'DBCursor') -> None:
-    """Fix kraken events with negative amounts related to staking ETH after the merge"""
-    log.debug('Enter _fix_kraken_eth2_events')
+def _fix_kraken_events(write_cursor: 'DBCursor') -> None:
+    """
+    Fix kraken events with negative amounts related to:
+    - staking ETH after the merge
+    - trades not having subtypes and having negative amounts
+    - withdrawals use positive amounts
+    - fix instant swaps
+    Needs to be executed after _update_history_events_schema
+    """
+    log.debug('Enter _fix_kraken_events')
+    log.debug('Fixing kraken eth2 related tuples')
+    update_tuples: list[Any] = []
     write_cursor.execute(
         'SELECT identifier, amount, usd_value FROM history_events WHERE location="B" AND '
         'asset=? AND type="staking" AND subtype="reward" AND CAST(amount AS REAL) < 0',
         (A_ETH2.identifier,),
     )
-    update_tuples = []
     for event_row in write_cursor:
+        amount = FVal(event_row[1])
+        usd_value = FVal(event_row[2])
         update_tuples.append(
             (
                 HistoryEventType.INFORMATIONAL.serialize(),
                 HistoryEventSubType.NONE.serialize(),
-                str(-FVal(event_row[1])),
-                str(-FVal(event_row[2])),
+                str(-amount) if amount != ZERO else str(amount),
+                str(-usd_value) if usd_value != ZERO else str(usd_value),
                 'Automatic virtual conversion of staked ETH rewards to ETH',
                 event_row[0],
             ),
         )
-
     if len(update_tuples) != 0:
         write_cursor.executemany(
             'UPDATE history_events SET type=?, subtype=?, amount=?, usd_value=?, notes=? WHERE identifier=?',  # noqa: E501
             update_tuples,
         )
-    log.debug('Exit _fix_kraken_eth2_events')
+
+    log.debug('Fixing kraken trades')
+    update_tuples = []
+    write_cursor.execute(
+        'SELECT identifier, amount, usd_value FROM history_events WHERE location="B" AND '
+        'type="trade" AND subtype="none"',
+    )
+    for event_row in write_cursor:
+        asset_amount = FVal(event_row[1])
+        usd_value = FVal(event_row[2])
```

Comment:
I don't want to do it like this because It would mean to trust the kraken data since the sign comes directly from them and we don't filter instead of trusting the types that is something that we deserialize